singleton/controller.go

```diff
@@ -36,3 +38,33 @@ func Source() source.Source {
 		},
 	})
 }
+
+func ChannelSource[T client.Object](ctx context.Context, objectChan <-chan watch.Event) source.Source {
+	eventSource := make(chan event.GenericEvent, 1)
```

Comment:
Should this channel buffer be larger? The writer is going to be forced to wait if the writer can't keep up so I wonder if we need to keep a larger channel buffer to make sure that we don't get stuck (it's pretty cheap from a memory perspective and should prevent our process from hanging on writes (maybe 1000?)

Reply:
Yeah, that seems reasonable